src/java.xml/share/classes/com/sun/org/apache/xerces/internal/utils/XMLSecurityManager.java

```diff
@@ -561,8 +556,42 @@ private boolean getSystemProperty(Limit limit, String sysPropertyName) {
                 states[limit.ordinal()] = State.SYSTEMPROPERTY;
                 return true;
             }
+        } catch (NumberFormatException e) {
+            //invalid setting
+            throw new NumberFormatException("Invalid setting for system property: " + limit.systemProperty());
+        }
```

Comment:
Kind of a questionable idiom here to catch NFE and then throw a new NFE, discarding the previous one. Maybe wrap instead? But the larger question is, I think, what should happen in this case. If the user sets a malformed property, will this cause exceptions to be thrown at arbitrary points in the program? Both items are things to consider for later.